app/src/main/java/ch/epfl/qedit/view/login/LogInActivity.java

```diff
@@ -0,0 +1,174 @@
+    }
+
+    /**
+     * Change app's language.
+     *
+     * @param languageCode the universal language code (e.g. "en" for English, "fr" for French)
+     */
+    private void setLanguage(String languageCode) {
+        context = LocaleHelper.setLocale(this, languageCode);
+        resources = context.getResources();
+    }
+
+    /** Update activity's texts (useful when the language is changed). */
+    private void updateTexts() {
+        emailField.setHint(resources.getString(R.string.hint_email));
+        passwordField.setHint(resources.getString(R.string.hint_password));
+        logInButton.setText(resources.getString(R.string.log_in_button_text));
+        Util.updateRedirectionText(resources, textViewSignUpInstead, R.string.sign_up_instead);
+
+        setTitle(resources.getString(R.string.title_activity_log_in));
+    }
+
+    /** Log in the user. */
+    private void logIn() {
+        // Check validity of each email and password before passing it to the Auth Service
+        Predicate<String> emailFormat = str -> str.matches(Util.REGEX_EMAIL);
+        Predicate<String> passwordFormat = str -> str.length() >= 6;
+        String email = Util.getString(emailField, emailFormat, resources, R.string.invalid_email);
+        String password =
+                Util.getString(passwordField, passwordFormat, resources, R.string.invalid_password);
+
+        // If the email or password is invalid, abort login
+        if (email == null || password == null) return;
+        // sanitize the email by removing starting and trailing space
+        email = email.trim();
+
+        progressBar.setVisibility(View.VISIBLE);
+        // Ask the authentication service for login
+        auth.logIn(email, password)
+                .whenComplete(
+                        (userId, error) -> {
+                            progressBar.setVisibility(View.GONE);
+                            if (error != null) onLogInFailed();
+                            else onLogInSuccessful(userId);
+                        });
+    }
+
+    private void onLogInSuccessful(String userId) {
+        Intent intent = new Intent(this, HomeActivity.class);
+        Bundle bundle = new Bundle();
+
+        DatabaseService db = DatabaseFactory.getInstance();
+        db.getUser(userId)
+                .whenComplete(
+                        (user, throwable) -> {
+                            if (throwable != null) {
+                                Util.showToast(R.string.database_error, context, resources);
+                            } else {
+                                bundle.putSerializable(USER, user);
+                                intent.putExtras(bundle);
+                                startActivity(intent);
+                            }
+                        });
+        // Put the current user id in cache
+        Util.putStringInPrefs(this, "user_id", userId);
```

Comment:
I suggest that we create a `static final` attribute in `login/Util.java` for `"user_id"`, same for the sign up counterpart.

Reply:
right i do this